test/link/glibc_compat/build.zig

```diff
@@ -17,4 +27,91 @@ pub fn build(b: *std.Build) void {
         _ = exe.getEmittedBin();
         test_step.dependOn(&exe.step);
     }
+
+    // Build & run against a sampling of supported glibc versions
+    for ([_][]const u8{
+        "native-linux-gnu.2.17", // Currently oldest supported, see #17769
+        "native-linux-gnu.2.23",
+        "native-linux-gnu.2.28",
+        "native-linux-gnu.2.33",
+        "native-linux-gnu.2.38",
+        "native-linux-gnu",
+    }) |t| {
+        const target = std.zig.CrossTarget.parse(
+            .{ .arch_os_abi = t },
+        ) catch unreachable;
+
+        const major_ver = target.toTarget().os.version_range.linux.glibc.major;
+        const minor_ver = target.toTarget().os.version_range.linux.glibc.minor;
+
+        std.debug.assert(major_ver == 2); // assuming no v1 or v3 to test
+
+        const exe = b.addExecutable(.{
+            .name = t,
+            .root_source_file = .{ .path = "glibc_runtime_check.zig" },
+            .target = target,
+        });
+        exe.linkLibC();
+
+        // Only try running the test if the host glibc is known to be good enough
+        if (minor_ver <= running_glibc_minor) {
+            const run_cmd = b.addRunArtifact(exe);
+            run_cmd.skip_foreign_checks = true;
+            run_cmd.expectExitCode(0);
+
+            test_step.dependOn(&run_cmd.step);
+        }
+
+        const check = exe.checkObject();
+
+        // __errno_location is always a dynamically linked symbol
+        check.checkStart();
+        check.checkInDynamicSymtab();
+        check.checkExact("0 0 UND FUNC GLOBAL DEFAULT __errno_location");
+
+        // before v2.32 fstatat redirects through __fxstatat, afterwards its a
+        // normal dynamic symbol
+        check.checkStart();
+        check.checkInDynamicSymtab();
+        if (minor_ver < 32) {
+            check.checkExact("0 0 UND FUNC GLOBAL DEFAULT __fxstatat");
+            //check.checkInSymtab();
+            //check.checkExact("{addr1} {size1} {shndx1} FUNC LOCAL HIDDEN fstatat");
```

Comment:
You should use `{addr}` for extracting values into variables that can later be compared using `checkComputeCompare`. Also, in order to extract values you should use `checkExtract`. Here, I think you wanted to use the same pattern you already used for `.dynsym` check, but a fuzzy check rather than exact match, so something like this:

```
check.checkInSymtab();
check.checkContains("FUNC LOCAL HIDDEN fstatat");
```

Reply:
Ah, makes sense!   Fixed. 

I've also added some `check.checkNotPresent` tests too ... do those look right?  (I'm not clear if they need their own `checkStart()`?)

Reply:
It depends. If you can guarantee that the next thing checked will be immediately after the previous check, `checkStart` is not needed. However, if that cannot be guaranteed, then there should be a `checkStart` before `checkNotPresent` call (or any other check for that matter).